Adding error boundary above App

Tamir198 opened the pull request:
## Changes Proposed

<!-- Describe the proposed changes and any additional information -->

<!-- Add all the screenshots which illustrate your changes -->

## Check List

<!-- Mark all the applicable boxes. To mark the box as done follow the following conventions -->
<!--
[x] - Correct; marked as done
[X] - Correct; marked as done

[ ] - Not correct; marked as **not** done
-->

- [x] The title of my pull request is a short description of the changes
- [x] This PR relates to some issue: <!-- use "Closes #999" to auto-close related issue -->
- [x] I have documented the changes made (if applicable)
- [ ] I have covered the changes with unit tests

Added global error boundary

Tamir198 commented:
@nir2002

Tamir198 commented:
@nir2002

undera commented on frontend/src/main.tsx:
```diff
@@ -2,9 +2,12 @@ import React from "react";
-  <React.StrictMode>
-    <App />
-  </React.StrictMode>
+  <ErrorBoundary fallback={<div>Something went wrong</div>}>
```
Showing "Something went wrong" is not enough. We should display some error details, as well as suggestion to reload the page or start from the root of the application.

Tamir198 replied:
I talked to nir regarding what more to do in this pr, I will have an update later this week

Tamir198 commented:
I added the global error modal component as a fallback for the error boundary. 

For testing, I added this line into the `App.tsx` file right above the render method: 

`  throw { title: "Error", message: "This is message" };` 

This an error thrown somewhere handled by the error boundary, if this is the case this is what you will see : 

![image](https://github.com/komodorio/helm-dashboard/assets/34707669/c8a2cd98-a529-4d5f-a4f7-1a217dee504c)

As you can see, right now I rely on the `{ title: "Error", message: "This is message" }` structure. 

@nir2002 

Is there any consistent structure that I can use for errors ? 

Ps -  Once we will have an error structure I will add prop type into the `ErrorBoundaryFallback`
